guides/source/security.md

```diff
@@ -80,31 +80,35 @@ The main objective of most attackers is to make money. The underground prices fo
+
+* Cookies imply a strict size limit of 4kB. Use the cookie only for data which is relevant for the session.
+
+* The client has full read and write access to the cookie. Client can copy cookies from computer to another. _In general, do not store anything sensitive or security related in the session._
+
+* The server can set expiration time for the cookie, but the client may delete the cookie and its contents before that. Also, the client may preserve cookie contents even for expired cookies. Persist the data you want to persist on the server side.
 
-Rails 2 introduced a new default session storage, CookieStore. CookieStore saves the session hash directly in a cookie on the client-side. The server retrieves the session hash from the cookie and eliminates the need for a session ID. That will greatly increase the speed of the application, but it is a controversial storage option and you have to think about the security implications of it:
+#### Security measures to protect your cookies
 
-* Cookies imply a strict size limit of 4kB. This is fine as you should not store large amounts of data in a session anyway, as described before. _Storing the current user's database id in a session is usually ok_.
+Setup secrets in config/secret.yml for all your environments.
 
-* The client can see everything you store in a session, because it is stored in clear-text (actually Base64-encoded, so not encrypted). So, of course, _you don't want to store any secrets here_. To prevent session hash tampering, a digest is calculated from the session with a server-side secret (`secrets.secret_token`) and inserted into the end of the cookie.
+When choosing secrets, make sure they are long and random enough, and consequently, harder to break. _Do not use a trivial secrets, i.e. a dictionary word, or secrets shorter than 30 characters_.
 
-However, since Rails 4, the default store is EncryptedCookieStore. With
-EncryptedCookieStore the session is encrypted before being stored in a cookie.
-This prevents the user from accessing and tampering the content of the cookie.
-Thus the session becomes a more secure place to store data. The encryption is
-done using a server-side secret key `secrets.secret_key_base` stored in
-`config/secrets.yml`.
+* When you have `secret_token` set, Rails is able to sign the cookie. The client can read the cookie, but cannot alter contents without Rails detecting it.
 
-That means the security of this storage depends on this secret (and on the digest algorithm, which defaults to SHA1, for compatibility). So _don't use a trivial secret, i.e. a word from a dictionary, or one which is shorter than 30 characters, use `rails secret` instead_.
+* When you have `secret_key_base` set, Rails is able to encrypt the cookie. The client cannot read or alter the cookie, without breaking the encryption.
```

Comment:
This and the above can be rephrased a bit: Rails refuses to start if it doesn't have a `secret_key_base`, so no need for conditionals.